AK/Time.h

```diff
@@ -354,6 +354,125 @@ class Duration {
     u32 m_nanoseconds { 0 }; // Always less than 1'000'000'000
 };
 
+namespace Detail {
+
+// Common base class for all naive time types. This class and its children are unaware of timezones.
+struct NaiveTime {
+    constexpr NaiveTime(NaiveTime const&) = default;
+    constexpr NaiveTime& operator=(NaiveTime const&) = default;
+
+    [[nodiscard]] timespec to_timespec() const { return m_offset.to_timespec(); }
+    // Rounds towards -inf.
+    [[nodiscard]] timeval to_timeval() const { return m_offset.to_timeval(); }
+
+    // We intentionally do not define a comparison operator here to avoid accidentally comparing incompatible time types.
+
+protected:
+    constexpr explicit NaiveTime(Duration offset)
+        : m_offset(offset)
+    {
+    }
+
+    Duration m_offset {};
+};
+
+}
+
+// Naive UNIX time, representing an offset from 1970-01-01 00:00:00Z, without accounting for UTC leap seconds.
+// This class is mainly intended for interoperating with anything that expects a unix timestamp.
+struct UnixDateTime : Detail::NaiveTime {
```

Comment:
"This class" (from the comment) is actually a struct - is there a reason why these are all `struct`s whereas we tend to use `class` for most other things in AK?

Reply:
It's a struct since it *should* be a POD or at least trivially copyable type.

Reply:
I agree that they feel like PODs, but I believe the common practice is that you move to a `class` when you start declaring constructors, methods, etc. I believe that to be the case for these structs as well.